collections/app/CollectionsComponents.scala

```diff
@@ -5,9 +5,7 @@ import play.api.ApplicationLoader.Context
 import router.Routes
 import store.CollectionsStore
 
-class CollectionsComponents(context: Context) extends GridComponents(context) {
-  final override lazy val config = new CollectionsConfig(configuration)
-
+class CollectionsComponents(context: Context) extends GridComponents(context, new CollectionsConfig(_)) {
```

Comment:
Huh, how does the `_` work? Does Scala work it out by type?

Reply:
It's currying. The type of the second param of `GridComponents[Config <: CommonConfig]` is `Configuration => Config` so we need to pass in a function that takes a play configuration and returns something that is a subclass of `CommonConfig`. `new CollectionsConfig(_)` is short for `c => new CollectionsConfig(c)` which is turning the constructor into a generic `Function1[Configuration, CollectionsConfig]` shape. The type argument of GridComponents is then inferred from the return value of the function we're passing in.

Reply:
> new CollectionsConfig(_) is short for c => new CollectionsConfig(c) 

Ah, this is what I was missing, thanks!